app/scripts/services/quota.js

```diff
@@ -254,6 +263,100 @@ angular.module("openshiftConsole")
+      "memory",
+      "requests.memory",
+      "limits.cpu",
+      "limits.memory"
+    ];
+
+    var getNotificaitonMessage = function(used, usedValue, hard, hardValue, quotaKey) {
+      var msgPrefix = "Your project is " + (hardValue < usedValue ? 'over' : 'at') + " quota. ";
+      var msg;
+      if (_.includes(COMPUTE_RESOURCE_QUOTAS, quotaKey)) {
+        msg = msgPrefix + "It is using " + percent((usedValue/hardValue), 0) + " of " + usageWithUnits(hard, quotaKey) + " " + humanizeQuotaResource(quotaKey, true) + ".";
+      } else {
+        msg = msgPrefix + "It is using " + usedValue + " of " + hardValue + " " + humanizeQuotaResource(quotaKey, true) + ".";
+      }
+
+      if (Constants.QUOTA_NOTIFICATION_MESSAGE && Constants.QUOTA_NOTIFICATION_MESSAGE[quotaKey]) {
+        msg += " " + Constants.QUOTA_NOTIFICATION_MESSAGE[quotaKey];
+      }
+
+      return msg;
+    };
+
+    // Return notifications if you are at quota or over any quota for any resource. Do *not*
+    // warn about quota for 'resourcequotas' or resources whose hard limit is
+    // 0, however.
+    var getQuotaNotifications = function(quotas, clusterQuotas, projectName) {
+      var notifications = [];
+
+      var notificationsForQuota = function(quota) {
+        var q = quota.status.total || quota.status;
+        _.each(q.hard, function(hard, quotaKey) {
+          var hardValue = usageValue(hard);
+          var used = _.get(q, ['used', quotaKey]);
+          var usedValue = usageValue(used);
+
+          // We always ignore quota warnings about being out of
+          // resourcequotas since end users cant do anything about it
+          if (quotaKey === 'resourcequotas' || !hardValue || !usedValue) {
+            return;
+          }
+
+          if(hardValue <= usedValue) {
+            var msg = getNotificaitonMessage(used, usedValue, hard, hardValue, quotaKey);
+            var containsHTML = hasHTML(msg);
+            notifications.push({
+              id: "quota-limit-reached-" + quotaKey,
+              namespace: projectName,
+              type: (hardValue < usedValue ? 'warning' : 'info'),
+              message: msg,
+              isHTML: containsHTML,
+              skipToast: true,
+              showInDrawer: true,
+              actions: [
+                {
+                  name: 'View Quotas',
+                  title: 'View project quotas',
+                  onClick: function(notification, action, index, group, drawer) {
+                    $location.url("/project/" + $routeParams.project + "/quota");
+                    drawer.drawerHidden = true;
+                  }
+                },
+                {
+                  name: "Don't Show Me Again",
+                  title: 'Permenantly hide this notificaiton until quota limit changes',
+                  onClick: function(notification, action, index, group, drawer) {
+                    NotificationsService.permanentlyHideNotification(notification.uid, notification.namespace);
```

Comment:
I don't think`notification.uid` is right. Where is `uid` getting set?

It should be something like

```js
var notificationID = 'openshift/hide-quote-warning/' + projectName + '/' + quotaKey;
```

Reply:
`notification.uid` is being set [here](https://github.com/openshift/origin-web-console/pull/2116/files#diff-e13814006e3a1d3d1f4fa5f45b6ebc5dR179)

Reply:
OK, I see. We should be scoping to a project. So maybe just above at line 313:

```js
var notificationID = 'quota-warning/' + projectName + '/' + quotaKey';
```

Then we can use that value here and don't need a reference to the notification itself.

Reply:
By the way, I don't think the NotificationDrawerWrapper is setting a `uid` anymore.

Reply:
common's [NotificationService](https://github.com/openshift/origin-web-common/blob/master/src/ui-services/notificationsService.js#L64) already has an API for permanently hiding a notification.  It takes an id and a 'namespace', which in this case is 'projectName'.

Reply:
Sorry, I see. That's fine then.

I think the point that the notification doesn't need to be passed back to the `onClick` handlers stands, though. We already have `quotaKey` and `projectName`.

Reply:
> By the way, I don't think the NotificationDrawerWrapper is setting a uid anymore.

I still see it being set in Ben's PR [here](https://github.com/openshift/origin-web-console/pull/2001/files#diff-e13814006e3a1d3d1f4fa5f45b6ebc5dR180)

Reply:
You're right. I don't think that's needed in Ben's PR -- we shouldn't need 3 separate IDs -- but I'll leave well enough alone since it's in the merge queue.

Reply:
3 IDs, wheeeee... 😄  
Yeah I agree, oops on my part, I prob should have dropped one after adding `trackByID`.